Makefile

```diff
@@ -0,0 +1,62 @@
+all: lint test check-coverage build  ## build, lint, test and check coverage
+	@[ -e .git/rebase-merge ] && git --no-pager log -1 --pretty='%h %s'
+	@echo '$(COLOUR_GREEN)Success$(COLOUR_NORMAL)'
+
+clean::  ## Remove generated files
+
+.PHONY: all clean
+
+# -- Build ---------------------------------------------------------------------
+
+BINARIES = $(notdir $(filter-out ./cmd/_%,$(wildcard ./cmd/*)))
+build: $(BINARIES)  ## Build binaries of directories in ./cmd
+
+ifneq ($(BINARIES),)
+$(BINARIES):
+	go build -o $@ ./cmd/$@
+
+clean::
+	rm -f $(BINARIES)
+
+.PHONY: build $(BINARIES)
+endif
+
+# -- Lint ----------------------------------------------------------------------
+
+lint:  ## lint the source code
+	golangci-lint run
+
+.PHONY: lint
+
+# -- Test ----------------------------------------------------------------------
+COVERFILE = coverage.out
+COVERAGE = 100
+
+test:  ## Run tests and generate a coverage file
+	go test -coverprofile=$(COVERFILE) ./...
+
+check-coverage: test  ## Check that test coverage meets the required level
+	@go tool cover -func=$(COVERFILE) | $(CHECK_COVERAGE) || $(FAIL_COVERAGE)
+
+cover: test  ## Show test coverage in your browser
+	go tool cover -html=$(COVERFILE)
+
+clean::
+	rm -f $(COVERFILE)
+
+CHECK_COVERAGE = awk -F '[ \t%]+' '/^total:/ && $$3 < $(COVERAGE) {exit 1}'
+FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_NORMAL)'; exit 1; }
+
+.PHONY: test check-coverage cover
+
+# --- Utilities ---------------------------------------------------------------
+
+COLOUR_NORMAL = $(shell tput sgr0 2>/dev/null)
+COLOUR_RED    = $(shell tput setaf 1 2>/dev/null)
+COLOUR_GREEN  = $(shell tput setaf 2 2>/dev/null)
+COLOUR_WHITE  = $(shell tput setaf 7 2>/dev/null)
+
+help:
+	@awk -F ':.*## ' 'NF == 2 && $$1 ~ /^[A-Za-z_-]+$$/ { printf "$(COLOUR_WHITE)%-30s$(COLOUR_NORMAL)%s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
```

Comment:
Add digits: `^[A-Za-z0-9_-]+`

Reply:
Done

Reply:
Done in bde16ab51f54405e2d35a0d96b149cd74c95e972